From c2cd5ee3410a1e5881c58dc8f9fe80fa719dbeac Mon Sep 17 00:00:00 2001
From: Wang ShaoBo <bobo.shaobowang@huawei.com>
Date: Thu, 30 Dec 2021 16:26:01 +0800
Subject: [PATCH] arm64/mpam: resctrl: add tips when rmid modification failed
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: c2cd5ee3410a1e5881c58dc8f9fe80fa719dbeac


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4LMMF
CVE: NA

-------------------------------------------------

This adds tips when rmid modification failed.

Fixes: a85aba6a1d67 ("mpam: Add support for group rmid modify")
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam/mpam_resctrl.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/kernel/mpam/mpam_resctrl.c b/arch/arm64/kernel/mpam/mpam_resctrl.c
index a9b99a0f347f..27b59c0fbf5a 100644
--- a/arch/arm64/kernel/mpam/mpam_resctrl.c
+++ b/arch/arm64/kernel/mpam/mpam_resctrl.c
@@ -1848,12 +1848,14 @@ static ssize_t resctrl_group_rmid_write(struct kernfs_open_file *of,
 
 	if (rmid == 0 || rdtgrp->mon.rmid == 0) {
 		ret = -EINVAL;
+		rdt_last_cmd_puts("default rmid 0 is always kept\n");
 		goto unlock;
 	}
 
 	ret = rmid_to_partid_pmg(rmid, &partid, &pmg);
 	if (ret < 0) {
 		ret = -EINVAL;
+		rdt_last_cmd_puts("invalid rmid\n");
 		goto unlock;
 	}
 
@@ -1862,6 +1864,7 @@ static ssize_t resctrl_group_rmid_write(struct kernfs_open_file *of,
 
 	if (rdtgrp->type != RDTCTRL_GROUP ||
 			!list_empty(&rdtgrp->mon.crdtgrp_list)) {
+		ret = -EINVAL;
 		rdt_last_cmd_puts("unsupported operation\n");
 		goto unlock;
 	}
-- 
2.26.2

